src/cmd/library_publish.js

```diff
@@ -66,32 +53,30 @@ export class LibraryPublishCommand extends Command {
 	 * @returns {Promise} To run the library publish command.
 	 */
 	run(state, site) {
-		const events = (event, ...args) => {
-			const fn = site[event].bind(site) || (() => {});
-			fn(...args);
-		};
+		let name;
+		return Promise.resolve(site.libraryIdent())
+			.then(ident => name = ident)
+			.then(() => site.apiClient())
+			.then((apiClient) => {
+				const promise = apiClient.publishLibrary(name)
```

Comment:
Suggestion: Maybe we can extract this to another function. It will reduce the 'width' of the code & make it a bit more readable.